dipy/reconst/sfm.py

```diff
@@ -458,21 +478,42 @@ def fit(self, data, mask=None):
         isotropic = self.isotropic(self.gtab).fit(data, mask)
         flat_params = np.zeros((data_in_mask.shape[0],
                                 self.design_matrix.shape[-1]))
+        del data_in_mask
+        gc.collect()
+
         isopredict = isotropic.predict()
         if mask is None:
             isopredict = np.reshape(isopredict, (-1, isopredict.shape[-1]))
         else:
             isopredict = isopredict[mask]
 
-        for vox, vox_data in enumerate(flat_S):
-            # In voxels in which S0 is 0, we just want to keep the
-            # parameters at all-zeros, and avoid nasty sklearn errors:
-            if not (np.any(~np.isfinite(vox_data)) or np.all(vox_data == 0)):
-                fit_it = vox_data - isopredict[vox]
-                with warnings.catch_warnings():
-                    warnings.simplefilter("ignore")
-                    flat_params[vox] = self.solver.fit(self.design_matrix,
-                                                       fit_it).coef_
+        if not num_processes:
+            num_processes = determine_num_processes(num_processes)
+
+        if num_processes > 1:
```

Comment:
`if num_processes > 1 and has_joblib:` instead

Reply:
Added